Content.Client/SubFloor/TrayScannerSystem.cs

```diff
@@ -130,36 +147,58 @@ private bool UpdateTrayScanner(EntityUid uid, TrayScannerComponent? scanner = nu
         // is still technically on
         if (flooredPos == Vector2.Zero)
         {
-            _subfloorSystem.ToggleSubfloorEntities(scanner.RevealedSubfloors, false, uid, _visualizerKeys);
+            _subfloorSystem.SetEntitiesRevealed(scanner.RevealedSubfloors, uid, false, _visualizerKeys);
             scanner.RevealedSubfloors.Clear();
             return true;
         }
 
+        // TODO REDO THIS
+        // Here we avoid the entity lookup & return early if the scanner's position hasn't appreciably changed. However,
+        // if a new player enters PVS-range, they will update the in-range entities on their end and use that to set
+        // LastLocation. This means that different players can technically see different entities being revealed by the
+        // same scanner. The correct fix for this is probably just to network the revealed entity set.... But I CBF
+        // doing that right now....
+        //
+        // Though this approach also has other issues. If the scanner is on a shuttle, and another grid moves close it,
+        // the scanner's local position doesn't update. But if you based this on the scanner's world position it would
+        // unnecessary update whenever the grid that it is on is moving. I guess you could just restrict scanners to
+        // only scan one grid at a time?
         if (flooredPos == scanner.LastLocation
             || (float.IsNaN(flooredPos.X) && float.IsNaN(flooredPos.Y)))
             return true;
 
         scanner.LastLocation = flooredPos;
 
-        // get all entities in range by uid
-        // but without using LINQ
+        //  Update entities in Range
         HashSet<EntityUid> nearby = new();
+        var coords = transform.MapPosition;
+        var worldBox = Box2.CenteredAround(coords.Position, (scanner.Range * 2, scanner.Range * 2));
+
+        foreach (var grid in _mapManager.FindGridsIntersecting(transform.MapID, worldBox, approx: true))
+        {
+            foreach (var entity in grid.GetAnchoredEntities(worldBox))
+            {
+                if (!Transform(entity).MapPosition.InRange(coords, scanner.Range))
+                    continue;
```

Comment:
I think this would get a different entity count depending on how the grid is rotated? If you use a Box2Rotated with the grid's rotation applied should work.

Reply:
The world-box should be big enough to contain the circle of radius `scanner.Range`, and the circular range check shouldn't care about rotations.

But in general yeah, a box lookup w/o considering rotations would be a problem, and was actually a problem in the previous implementation